### PR TITLE
Resolve dependabot PRs

### PR DESCRIPTION
**Pull request by alanssitis**
Dependabot PRs #68 and #69 are not progressing since the bot is not rebasing it.

This PR basically:
- Update package requirements
- Bump actions/checkout versions
- Fix clippy (linter) errors: Deprecated and Updated APIs
- Removed unnecessary actions
- Replaced archived actions

API generating new ECDSA pair requires an RNG, passing `ring`'s [`SystemRandom`](https://docs.rs/ring/latest/ring/rand/struct.SystemRandom.html).

Should close the PRs above: #68, #69

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -9,42 +9,42 @@ jobs:
   check:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
```
Comment (adityasaky) [inline .github/workflows/rust.yml:12]: any reason not to use the latest release?
```suggestion
    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
```

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -9,42 +9,42 @@ jobs:
...
     - name: Check
       run: cargo check
   test:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
```
Comment (adityasaky) [inline .github/workflows/rust.yml:18]: same here

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -9,42 +9,42 @@ jobs:
...
     - name: Run tests
       run: cargo test --verbose
   fmt:
     name: Rustfmt
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
```
Comment (adityasaky) [inline .github/workflows/rust.yml:25]: and here

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -9,42 +9,42 @@ jobs:
...
         with:
           command: fmt
           args: --all -- --check
   clippy:
     name: Run Clippy
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
```
Comment (adityasaky) [inline .github/workflows/rust.yml:40]: and here

File: .github/workflows/rust.yml [hunk 5]
```diff
@@ -9,42 +9,42 @@ jobs:
...
+      - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af
```
Comment (adityasaky) [inline .github/workflows/rust.yml:41]: This repository has been archived, can we find something that's still maintained?

File: .github/workflows/rust.yml [hunk 6]
```diff
@@ -9,42 +9,42 @@ jobs:
...
         with:
           profile: minimal
           toolchain: stable
           override: true
       - run: rustup component add clippy
-      - uses: actions-rs/cargo@v1
+      - uses: actions-rs/cargo@844f36862e911db73fe0815f00a4a2602c279505
```
Comment (adityasaky) [inline .github/workflows/rust.yml:47]: Also archived, is there native support in actions now?

File: src/crypto.rs [hunk 7]
```diff
@@ -593,9 +593,12 @@ impl PrivateKey {
         der_key: &[u8],
         scheme: SignatureScheme,
     ) -> Result<Self> {
-        let key_pair =
-            EcdsaKeyPair::from_pkcs8(&ECDSA_P256_SHA256_ASN1_SIGNING, der_key)
-                .unwrap();
+        let key_pair = EcdsaKeyPair::from_pkcs8(
+            &ECDSA_P256_SHA256_ASN1_SIGNING,
+            der_key,
+            &SystemRandom::new(),
```
Comment (adityasaky) [inline src/crypto.rs:599]: can you clarify this change in the PR desc?

File: src/crypto.rs [hunk 8]
```diff
@@ -569,10 +569,10 @@ impl PrivateKey {
             Error::Encoding("Could not parse key as PKCS#8v2".into())
         })?;
 
-        if key.public_modulus_len() < 256 {
+        if key.public().modulus_len() < 256 {
```
Comment (adityasaky) [inline src/crypto.rs:572]: deprecation warning fix?

**Review by adityasaky — CHANGES_REQUESTED** [review]
Thanks for helping close those out! I have some comments / questions.

File: src/crypto.rs (hunk 7 above)
Comment (alanssitis) [reply src/crypto.rs:599]: Done

File: src/crypto.rs (hunk 8 above)
Comment (alanssitis) [reply src/crypto.rs:572]: Yup, added to PR description.

File: .github/workflows/rust.yml (hunk 5 above)
Comment (alanssitis) [reply .github/workflows/rust.yml:41]: I was working on something in #76, will move it here.

File: src/crypto.rs (hunk 7 above)
Comment (adityasaky) [reply src/crypto.rs:599]: Did the API change to require passing in an RNG? Is this a cryptographically safe RNG to use here?

Comment (alanssitis) [reply src/crypto.rs:599]: Yes. It is also cryptographically safe; the same API generates new keys and sign objects.

https://docs.rs/ring/latest/ring/rand/struct.SystemRandom.html

**Review by adityasaky — APPROVED** [review]
Thanks!

**Pull request merged**